demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py

```diff
@@ -311,44 +339,59 @@ def verify_results_against_test_data(
-        for key, val in expected_values.items():
-            if val:
-                result_val = result.get(key)
-                out = f"Checking for key {key}:\n - expected: {val}\n - received: {result_val}"
+        for expected_key, expected_value in expected_values.items():
+            if expected_value:
+                received_value_before = result.get(expected_key)
+                expected_value_type = get_type_pretty_name(expected_value)
+                (
+                    received_value_type,
+                    received_value,
+                ) = transform_received_value_by_expected_type(
+                    received_value_before, expected_value_type
+                )
+                out = (
+                    f"Checking for key {expected_key} - "
+                    f"expected:{expected_value} expected type:{expected_value_type} "
+                    f"received:{received_value} received type:{received_value_type} "
+                    f"before transform by expected - received:{received_value_before} received type: "
+                    f"{get_type_pretty_name(received_value_before)}"
+                )
                 logger.debug(f"[cyan]{out}[/cyan]", extra={"markup": True})
                 result_test_case_system_out.append(out)
-
-                if type(result_val) == type(val) and result_val == val:
-                    out = f"Value:{result_val} and Type:{get_type_pretty_name(result_val)} Matched for key {key}"
+                if (
+                    received_value == expected_value
+                    and received_value_type == expected_value_type
+                ):
+                    out = f"Value:{received_value} and Type:{received_value_type} Matched for key {expected_key}"
                     result_test_case_system_out.append(out)
                     logger.debug(out)
                 else:
-                    if type(result_val) == type(val):
+                    if received_value_type == expected_value_type:
                         err = (
-                            f"Expected value does not match for key {key}: - expected: {val} - received: {result_val} "
-                            f"Types match:{get_type_pretty_name(result_val)}"
+                            f"Expected value does not match for key {expected_key}: - expected: {expected_value} - "
+                            f"received: {received_value} Types match:{received_value_type}"
                         )
                         logger.error(
-                            f'[red][bold]{key}[/bold] --- "{result_val}" != "{val}" '
-                            f"Types match:{get_type_pretty_name(result_val)}[/red]",
+                            f'[red][bold]{expected_key}[/bold] --- "{received_value}" != "{expected_value}" '
+                            f"Types match:{received_value_type}[/red]",
                             extra={"markup": True},
                         )
                     else:
```

Comment:
Document this `else`. Something like:
```suggestion
                    else:
                        # Types don't match, so values are not matching either,
                        # so it means that both do not match.
```